pat/consumer_tests/step_6.py

```diff
@@ -0,0 +1,469 @@
+"""Tests operations."""
+
+from __future__ import annotations
+
+import decimal
+import functools
+import logging
+import random
+import string
+import threading
+import typing
+import uuid
+
+# result_collector removed; use logger and boolean returns instead
+from sdc11073.mdib import descriptorcontainers, statecontainers
+from sdc11073.observableproperties import observables
+from sdc11073.xml_types import msg_types, pm_qnames, pm_types
+
+if typing.TYPE_CHECKING:
+    from collections.abc import Sequence
+    from concurrent.futures import Future
+
+    from sdc11073.consumer import SdcConsumer, operations
+    from sdc11073.consumer.serviceclients import contextservice, setservice
+
+__STEP__ = '6'
+logger = logging.getLogger('pat.consumer')
+
+
+def test_6b(consumer: SdcConsumer) -> bool:  # noqa: C901
+    """The Reference Consumer invokes SetContextState:
+    - Payload: 1 patient context
+    - Context state is added to the MDIB including context association and validation
+    - If there is an associated context already, the Reference Provider disassociates that context
+    - Handle and version information is generated by the Reference Provider
+    - In order to avoid infinite growth of patient contexts, older contexts are allowed to be removed from the MDIB (when ContextAssociation is No)
+    - The Reference Provider successfully concludes the invocation
+    """  # noqa: D205, D400, D415, E501, W505
+    set_context_state_handle = 'set_context_0.sco.mds_0'
```

Comment:
The gandle of the operation must not be hardcoded here.
In my opnion the operation to use should be identified by its code.